pkg/controller/pods.go

```diff
@@ -43,3 +47,26 @@ func (c *Controller) handlePodChange(obj interface{}) {
 	key := fmt.Sprintf("%s/%s", object.GetNamespace(), instanceName)
 	c.healthCheckQueue.AddAfter(key, 1*time.Second)
 }
+
+// DeletePodsByStatefulSet deletes all pods associated with a statefulset
+func (c *Controller) DeletePodsByStatefulSet(ctx context.Context, sts *appsv1.StatefulSet) (err error) {
```

Comment:
Unfortunately, there is no `Recreate` update strategy for statefulsets, otherwise we could have use the `kubectl rollout restart` method and let all pods recreate itself without manual deleting them. Not sure why Kubernetes doesn't provide this functionality for a statefulset, but it does provide it for normal deployments.

Reply:
That what the `Recreate` update strategy would solve, but it's not supported for a statefulset (only for deployments), so it's not an option here.